src/node_url.cc

```diff
@@ -492,7 +492,9 @@ namespace url {
     if (flags->IsInt32())
       base->flags = flags->Int32Value(context).FromJust();
 
-    GET_AND_SET(env, base_obj, scheme, base, URL_FLAGS_HAS_SCHEME);
+    Local<Value> scheme = GET(env, base_obj, "scheme");
+    base->scheme = Utf8Value(env->isolate(), scheme).out();
+
     GET_AND_SET(env, base_obj, username, base, URL_FLAGS_HAS_USERNAME);
```

Comment:
Eventually it would likely be better just to inline this and get rid of the macro